MRFIXIT: When the graphics stack is stuck, we can't restart using S31emulationstation. Need to killall and then restart using a separate script.
--- a/es-app/src/SystemData.cpp	2018-07-10 03:42:13.000000000 -0400
+++ b/es-app/src/SystemData.cpp	2018-07-25 19:38:05.013628499 -0400
@@ -227,7 +227,16 @@
   }
 
   // Reinit
-  window->init();
+  if(!window->init())
+  {
+        // Failed to re-init the renderer. Sometimes emulators don't clean up completely after themselves and hold on to resources that ES needs.
+        // So let's restart ES, which will clear that out and create a new context. 
+        LOG(LogError) << "Forcing a restart of ES to clear the stuck graphics renderer...";
+        command = "/recalbox/scripts/es_restart.sh";
+        system(command.c_str());
+        return;
+  }
+
   VolumeControl::getInstance()->init();
   AudioManager::getInstance()->resumeMusic();
   window->normalizeNextUpdate();
